src/relax/op/op.cc

```diff
@@ -331,8 +331,133 @@ RELAY_REGISTER_OP("relax.call_tir")
     .set_attr<FNormalize>("FNormalize", NormalizeCallTIR)
     .set_attr<Bool>("FPurity", Bool(true));
 
-Expr MakeCallTIR(Expr func, Tuple args, Array<TensorStructInfo> out_sinfo_list,
+static Array<TensorStructInfo> InferCallTIROutputStructInfo(Expr func, Tuple args,
```

Comment:
One thing to note is that this is not always possible to do such inference. Since it is possible to have tir functions like reshape, where the output shape is being explicitly specified via the destination. For the particular low-level call_tir op. I think it is safer to always ask for the sinfo, then explicitly checks the consistency to avoid error

Reply:
Absolutely agreed that we should check for consistency after generating the IR, and that's something I want to add to the well-formed checker as well.  This specific PR would be to avoid inconsistency while generating the IR.

(And if we can't infer the output shape, then the output shape must still be be explicitly provided.)